### PR TITLE
Fix DeleteMemberCluster clusterID error

### PR DESCRIPTION
**Pull request by kebe7jun**

### CODE REVIEW

**Review by hzxuzhonghu — APPROVED** [review]
good catch

File: pilot/pkg/serviceregistry/kube/controller/multicluster.go [hunk 1]
```diff
@@ -155,9 +155,11 @@ func (m *Multicluster) close() (err error) {
 	// Remove all of the clusters.
 	g, _ := errgroup.WithContext(context.Background())
 	for _, clusterID := range clusterIDs {
-		g.Go(func() error {
-			return m.DeleteMemberCluster(clusterID)
-		})
+		func(cid string) {
```
Comment (howardjohn) [inline pilot/pkg/serviceregistry/kube/controller/multicluster.go:158]: prefer `clusterID:=clusterID` to avoid needing to nest it more

Comment (ramaraochavali) [reply pilot/pkg/serviceregistry/kube/controller/multicluster.go:158]: +1. https://github.com/istio/istio/pull/32225

**Pull request merged**